dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -2462,50 +2464,49 @@ RtpsUdpDataLink::bundle_mapped_meta_submessages(
 }
 
 void
-RtpsUdpDataLink::enable_response_queue()
+RtpsUdpDataLink::flush_send_queue(const MonotonicTimePoint& /*now*/)
 {
-  ACE_GUARD(ACE_Thread_Mutex, g, send_queues_lock_);
-  ThreadSendQueueMap::iterator it = thread_send_queues_.find(ACE_Thread::self());
-  if (it == thread_send_queues_.end()) {
-    thread_send_queues_[ACE_Thread::self()] = make_rch<SubmessageQueue>();
-  }
-}
+  MetaSubmessageVecVec temp;
 
-void
-RtpsUdpDataLink::disable_response_queue()
-{
-  SubmessageQueue_rch queue;
   {
-    ACE_GUARD(ACE_Thread_Mutex, g, send_queues_lock_);
-    ThreadSendQueueMap::iterator it = thread_send_queues_.find(ACE_Thread::self());
-    if (it != thread_send_queues_.end()) {
-      queue = it->second;
-      thread_send_queues_.erase(it);
-    }
-  }
-  if (queue && !queue->empty()) {
-    bundle_and_send_submessages(*queue);
+    ACE_GUARD(ACE_Thread_Mutex, g, send_queue_lock_);
+    temp.swap(send_queue_);
+    total_submessage_count_ = 0;
   }
+
+  bundle_and_send_submessages(temp);
 }
 
 void
 RtpsUdpDataLink::queue_or_send_submessages(MetaSubmessageVec& in)
 {
-  {
-    ACE_GUARD(ACE_Thread_Mutex, g, send_queues_lock_);
+  if (in.empty()) {
+    return;
+  }
 
-    ThreadSendQueueMap::iterator it = thread_send_queues_.find(ACE_Thread::self());
-    if (it != thread_send_queues_.end()) {
-      it->second->push_back(MetaSubmessageVec());
-      it->second->back().swap(in);
-      return;
+  MetaSubmessageVecVec temp;
+
+  {
+    ACE_GUARD(ACE_Thread_Mutex, g, send_queue_lock_);
+
+    send_queue_.push_back(MetaSubmessageVec());
+    send_queue_.back().swap(in);
+    total_submessage_count_ += send_queue_.back().size();
+
+    // Heartbeats and Acknacks are around 8 longs.
```

Comment:
Worst case Gap and AckNack are about 16.

Reply:
I'm not sure where to put my comment, so I'll just add it here:
- I worry that this PR is removing functionality (the ability to temporarily enforce bundling for a group of actions taken by a thread (e.g. in response to a single packet or in response to a single timer)) which might prove beneficial later.
- I worry that the only division in packets now is time & size, meaning our Wireshark output might start to look a whole lot more "jumbled".
- If we're going to go this route, I would suggest we at least have separate queues for readers & writers for the sake of making packet inspection less chaotic.